openraft/src/replication/mod.rs

```diff
@@ -751,7 +756,13 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> Replication
                 "sending snapshot chunk"
             );
 
-            let res = timeout(self.install_snapshot_timeout, self.network.send_install_snapshot(req)).await;
+            let snap_timeout = if done {
+                self.install_snapshot_timeout
+            } else {
+                self.finalize_snapshot_timeout
+            };
```

Comment:
Should the timeouts be in opposite places?

Reply:
yes you're right.

Reply:
> Both `install_snapshot_timeout` and `finalize_snapshot_timeout` sound like the final part of installing a snapshot. What about 'send_snapshot_timeout` for the new timeout name?

so `send_snapshot_timeout` sounds like what would be used for either a timeout for the entire snapshot process or for one of the parts which is  `install_snapshot_timeout`. How about switching to `send_snapshot_timeout` but use it for the parts and then using `install_snapshot_timeout` for the last part?